backend/tests/Ragas/utils/modules/ragas_evaluation.py

```diff
@@ -145,61 +145,51 @@ async def evaluate_with_ragas(
         print("Running RAGAS evaluation (this may take a while)...")
         results = evaluate(dataset=dataset, metrics=metrics, llm=llm)
 
-        # Define the expected metrics we want in our final output
-        expected_metrics = ["factual_correctness", "semantic_similarity", "answer_accuracy"]
-
-        # Create a base DataFrame for our results
-        result_data = []
-
-        # Create results for each sample with placeholder metric values
-        for i, sample in enumerate(samples):
-            result_data.append(
-                {
-                    "question": sample.user_input,
-                    **{metric: None for metric in expected_metrics},  # Initialize all metrics as None
-                }
-            )
-
         try:
-            if hasattr(results, "to_pandas"):
-                scores_df = results.to_pandas()
-                available_columns = list(scores_df.columns)
-                print(f"Results DataFrame columns: {available_columns}")
-
-                # Map available columns to expected metrics
-                for i, (_, row) in enumerate(scores_df.iterrows()):
-                    if i < len(result_data):
-                        for col in available_columns:
-                            matching_metric = next(
-                                (m for m in expected_metrics if m in col.lower() or col.lower() in m), None
-                            )
-                            if matching_metric and i < len(result_data):
-                                result_data[i][matching_metric] = row[col]
+            print("Processing evaluation results...")
+            # Define column mappings for RAGAS output to our desired output format
+            columns = {
+                "nv_accuracy": "answer_accuracy",
+                "factual_correctness(mode=f1)": "factual_correctness",
+                "semantic_similarity": "semantic_similarity",
+                "user_input": "question",
+            }
+
+            # Print available columns for debugging
+            available_columns = list(results.to_pandas().columns)
+            print(f"Results DataFrame columns: {available_columns}")
+
+            # Check that all required columns exist
+            missing_columns = [col for col in columns.keys() if col not in available_columns]
+            if missing_columns:
+                raise ValueError(
+                    f"Missing expected columns in RAGAS output: {missing_columns}. Update column mappings."
+                )
+
+            # Create DataFrame with only needed columns and renamed according to our convention
+            full_df = results.to_pandas()
+            selected_df = full_df[list(columns.keys())]
+            # Manually rename columns to avoid pyright issues
```

Comment:
I've done a bit of research into why this is being done - I really don't like the approach to have to make a copy of data just because of type-hinting problems, but it seems like the approach you've taken is a common pattern to am happy to live with it.

It took me a little while to understand the reasoning behind what was going on, so I've proposed a slightly different error message
```suggestion
            # Copy to guarantee standalone DataFrame as opposed to view
            # See https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
```